src/middlewares/validate.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * @param {*} schema 
+ * @returns{object} allows next() if the request body is valid otherwise returns an error
+ */
+export default function validate(schema) {
+  return (req, res, next) => {
+    const { error } = schema.validate(req.body);
+    if (error) {
+      return res.status(400).send({
+        status: 400,
+        data: { error: error.details[0].message.replace(/[/"]+/g, '') }
```

Comment:
the error object can have the string message and the full error message. its very use full as well in debugging or adding context to the client side